File: nga_phone_base_3.0/src/main/java/sp/phone/ui/fragment/dialog/MakeNoteDialogFragment.java
package sp.phone.ui.fragment.dialog;

import android.app.AlertDialog;
import android.app.Dialog;
import android.os.Bundle;
import android.view.View;
import android.widget.CheckBox;
import android.widget.EditText;

import java.lang.reflect.Field;
import gov.anzong.androidnga.base.util.ToastUtils;
import gov.anzong.androidnga.R;
import sp.phone.util.NLog;

public class MakeNoteDialogFragment extends BaseDialogFragment {

    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {

        final View view = getActivity().getLayoutInflater().inflate(R.layout.dialog_make_note, null);
        AlertDialog.Builder builder = new AlertDialog.Builder(getContext());
        EditText input = view.findViewById(R.id.make_note_edit_text);
        builder.setTitle(R.string.mark_note)
                .setView(view)
                .setPositiveButton("确认", (dialog, whichButton) -> {
                    int length = input.getText().toString().length();
                    if (length > 1 && length < 20) {
                        String prefix = getArguments().getString("prefix");
                    } else {
                        ToastUtils.error("备注内容长度必须在1~20字范围内");
                    }
                    try {
                        Field field = dialog.getClass().getSuperclass()
                                .getDeclaredField("mShowing");
                        field.setAccessible(true);
                        field.set(dialog, false);
                    } catch (Exception e) {
                        e.printStackTrace();
                    }
                }).setNegativeButton(android.R.string.cancel,
                        (dialog, whichButton) -> {
                            dialog.dismiss();
                            try {
                                Field field = dialog.getClass().getSuperclass()
                                        .getDeclaredField("mShowing");
                                field.setAccessible(true);
                                field.set(dialog, true);
                            } catch (Exception e) {
                                e.printStackTrace();
                            }
                        });
        return builder.create();
    }
}

File: nga_phone_base_3.0/src/main/java/sp/phone/ui/fragment/dialog/MakeNoteDialogFragment.kt
package sp.phone.ui.fragment.dialog;

import android.app.AlertDialog
import android.app.Dialog
import android.content.DialogInterface
import android.os.Bundle
import android.view.View
import android.widget.EditText
import gov.anzong.androidnga.R
import gov.anzong.androidnga.base.util.ToastUtils
import sp.phone.common.NoteInfo
import sp.phone.common.NoteManangerImpl.Companion.getInstance


class MakeNoteDialogFragment : BaseDialogFragment() {
    override fun onCreateDialog(b: Bundle?): Dialog {
        val view: View = requireActivity().layoutInflater.inflate(R.layout.dialog_make_note, null)
        val builder = AlertDialog.Builder(context)
        val userName = requireArguments().getString("userName")
        val uid = requireArguments().getString("uid")
        val input = view.findViewById<EditText>(R.id.make_note_edit_text)
        val noteMananger = getInstance()
        val noteStr = uid?.let { noteMananger?.getNoteFromList(it) }
        input.setText(noteStr);
        builder.setTitle(R.string.mark_note)
            .setView(view)
            .setPositiveButton(
                "确认"
            ) { dialog: DialogInterface, _: Int ->
                val length = input.text.toString().length
                if (length in 2..19) {

                    val note = input.text ?: ""
                    val noteInfo = NoteInfo()
                    noteInfo.note = note.toString()
                    noteInfo.userId = uid!!
                    noteInfo.nickName = userName
                    noteMananger?.addToNotesList(noteInfo)
                    dialog.dismiss()
                    ToastUtils.success(R.string.add_to_noteslist_success);
                } else {
                    ToastUtils.error("备注内容长度必须在1~20字范围内")
                }
                try {
                    val field = dialog.javaClass.superclass
                        .getDeclaredField("mShowing")
                    field.isAccessible = true
                    field[dialog] = false
                } catch (e: Exception) {
                    e.printStackTrace()
                }
            }.setNegativeButton(
                R.string.cancel,
                DialogInterface.OnClickListener { dialog: DialogInterface, whichButton: Int ->
                    dialog.dismiss()
                    try {
                        val field = dialog.javaClass.superclass
                            .getDeclaredField("mShowing")
                        field.isAccessible = true
                        field[dialog] = true
                    } catch (e: Exception) {
                        e.printStackTrace()
                    }
                })
        return builder.create()
    }

}
